package.json

```diff
@@ -4,15 +4,16 @@
   "description": "React-fiber renderer for THREE.js",
   "main": "dist/index.cjs.js",
   "module": "dist/index.js",
+  "types": "types/index.d.ts",
   "sideEffects": false,
   "scripts": {
     "prebuild": "rimraf dist",
     "build": "rollup -c",
-    "prepare": "npm run build",
+    "prepare": "npm run build && npm run typegen",
```

Comment:
`prepare` will now run both the build and typegen before publishing to npm

Reply:
this doesn't let me publish though, it creates the typefiles but any error before publish makes the process stop.

Reply:
ahh fair, hmmmm will think about this one and if we can swallow the exit codes in the meantime.